### PR TITLE
Editor: report warnings about possibly unlinked event handlers

### PR DESCRIPTION
**Pull request by ivan-mogilko**
As suggested by @messengerbag, as a most primitive solution for #2550.
...and a complementary to #2556

This adds another type of post-compilation warning that reports a script function that *looks like* it's a event handler, but *not linked* to the object events.

The feature uses default function names, as configured in Interaction tables.
For example: `cEgo_Look`, `Room_Load` and so forth.

The reported warning uses following pattern:
```Function "ObjectName_EventName" looks like an event handler, but is not linked on ObjType (ID) ScriptName's Event pane```

Example:
warnings--eventhandlers

**TODO:**
Option in Editor Preferences that turns this on and off?

Need to decide what happens if user double clicks the warning. Do we want it to go to the script, or to the object's Events Pane?
Or have both options, called from a context menu?
See also: #2589

Unfortunately, Autocomplete data, which we use for this kind of scanning, contains the index of the function's *starting character*, rather than a *line number*. I think the autocomplete may be amended to also have line numbers for script tokens.

### CODE REVIEW

**Comment by ericoporto** [comment]
This is great! I think an option in Editor Preferences to turn this off is fine, just important to make sure the default for this is on. If this check is thought as being a build setting, then perhaps it's best that is done in the general settings in the build section.

**Comment by ivan-mogilko** [comment]
Added following:
* Double clicking on "unlinked function" warning zooms into the script file. I was conflicted here about opening an object's event tab, because maybe user would like to see the function itself first, before deciding what to do with it?
* Double clicking on "missing function" warning zooms (at least tries to) into the event tab where this function is assigned.
In order to do this I had to move the parts of "Go to Definition" command code from ScriptEditorBase to GUIController.
Also changed `ShowItemPaneByName` to have a bool return type.

**Comment by ivan-mogilko** [comment]
On a side note, first I checked if it's going to be quick to add "line number" to function entry in AutoComplete, but no. Unfortunately, AutoComplete parser simply does not have a record of "current line number". So adding this would require more work on its own.

On another hand, apparently the "SpeechCenter" plugin does NOT use AGS Editor's AutoCompleteData, as I suspected previously, but has its own script parser. I don't know many editor plugins, and from the ones I heard of this is probably the only one that does thorough script parsing. So idk if there's any editor plugins that would access AutoCompleteData, maybe not.

There's currently a problem with AutoComplete entry classes (ScriptToken and descendants) not using *properties*. They have bare public fields in them, which is really bad for a public interface.
I'm not an expert in C#, and don't quite know how it works in case of dynamic assembly linking (as in case of editor->plugin link).
By experiment I found that if you have a property used by plugin, then *order* of this property inside a class does not matter. This seems different from, for example, C++, where moving a method inside an interface class could change order of entries in a virtual table, and break things. *Maybe* in C# all of these methods and properties are automatically loaded by names, similar to loading a function address by name in C++.

Anyway, I am saying all this, because I think the AutoComplete token classes must be modified to have readonly properties in them, and have the regular fields made private, to secure them from direct access by plugins.

**Comment by ivan-mogilko** [comment]
I went a little further, and moved message-specific behavior to OutputPanelItem class, which is a descendant of ListViewItem. This class wraps CompileMessage, and implements interaction with it.
Initially I had an idea to have subclasses per a type of CompileMessage, but decided to keep a single "base" class for the time being.

OutputPanelItem returns a list of Actions, which OutputPanel fills into the context menu.
There are 2 methods: Action(string command) and DefaultAction(). First is called when user selects a menu item, and second is called whenever user double-clicks an item.

In practice:
- a "Missing handler" message shows "Go to Object" menu command;
- a "Possibly unlinked handler" message shows both "Go to Script" and "Go to Object" menu commands.

**Comment by ericoporto** [comment]
This works, but I think there is some weird interaction in case of room scripts. If a room script has like an oObject_Interact and it is not linked to the object, it will show the message, double clicking it will go to the script - I understand, I saw the issue to address the problem separately - but the big issue is that it then clears all warning messages. I think this is just some other bug in AGS Editor, I believe this is part of #1439.

So in summary, it looks like this works, but particularly for room scripts, it is hard to follow reported warnings. But this looks very useful, and the issue looks like something that can be fixed.

**Comment by ivan-mogilko** [comment]
> it will show the message, double clicking it will go to the script - I understand, I saw the issue to address the problem separately - but the big issue is that it then clears all warning messages.

Do you mean that it clears instantly after opening a room script, or after some other action from your side?

**Comment by ericoporto** [comment]
Ah, I understand now, it is that I instinctively save after editing. I always ctrl+s after editing a file, which makes the room file rebuild, which then clears all warnings.

https://github.com/user-attachments/assets/41aa0340-da19-4e04-98cd-602dce3af37f

**Comment by ivan-mogilko** [comment]
> Ah, I understand now, it is that I instinctively save after editing. I always ctrl+s after editing a file, which makes the room file rebuild, which then clears all warnings.

Hmm, this behavior on video looks strange.
So you are in a room script of a room which had warnings reported for, you save, but all the warnings disappear, including the warnings related to that room.
I can see how it may be annoying that all the other room warnings disappear. That may a be an issue, but it's related to the output panel behavior.

But I think that *current room* warnings should reappear again, since you did not fix them and saved again. Probably the handler checks are skipped in some case.

**Comment by ivan-mogilko** [comment]
> Do I remember right that in ags3 rooms are recompiled when their scripts are saved?

So, that's where this issues comes from. Saving just the room script does *not* recompile neither the script nor the room, and yet the output pane is cleared.

But furthermore, there's no such thing as "save room script" or "save script" in AGS Editor. There's just "Save" that saves whole game... That's why it clears the output.

**Comment by ivan-mogilko** [comment]
Well, it seems that the remaining issues are general issues of the rooms and output panel, so i'll merge this feature.
